kv/etcd/etcd.go

```diff
@@ -269,6 +301,80 @@ func (c *client) CheckAndSet(key string, version int, v proto.Message) (int, err
 	return version + 1, nil
 }
 
+func (c *client) getCache(key string) (kv.Value, bool) {
+	c.cache.RLock()
+	v, ok := c.cache.Values[key]
+	c.cache.RUnlock()
+
+	return v, ok
+}
+
+func (c *client) mergeCache(key string, v *value) {
+	c.cache.Lock()
+	defer c.cache.Unlock()
+
+	cur, ok := c.cache.Values[key]
+	if !ok || v.isNewer(cur) {
+		c.cache.Values[key] = v
+		go c.syncFS()
+	}
+}
+
+func (c *client) syncFS() error {
```

Comment:
In that case maybe don't return anything since you already logged and emit metric inside the function itself?